pkg/cliplugins/workload/plugin/sync.go

```diff
@@ -284,14 +303,113 @@ func (o *SyncOptions) enableSyncerForWorkspace(ctx context.Context, config *rest
+	oldData, err := json.Marshal(workloadv1alpha1.SyncTarget{
+		Spec: workloadv1alpha1.SyncTargetSpec{
+			SupportedAPIExports: syncTarget.Spec.SupportedAPIExports,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to Marshal old data for syncTarget %s: %w", syncTargetName, err)
+	}
+
+	newData, err := json.Marshal(workloadv1alpha1.SyncTarget{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:             syncTarget.UID,
+			ResourceVersion: syncTarget.ResourceVersion,
+		}, // to ensure they appear in the patch as preconditions
+		Spec: workloadv1alpha1.SyncTargetSpec{
+			SupportedAPIExports: supportedAPIExports,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to Marshal new data for syncTarget %s: %w", syncTargetName, err)
+	}
+
+	patchBytes, err := jsonpatch.CreateMergePatch(oldData, newData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create merge patch for syncTarget %q because: %w", syncTargetName, err)
+	}
+
+	if syncTarget, err = kcpClient.WorkloadV1alpha1().SyncTargets().Patch(ctx, syncTargetName, types.MergePatchType, patchBytes, metav1.PatchOptions{}); err != nil {
+		return nil, fmt.Errorf("failed to patch syncTarget %s: %w", syncTargetName, err)
+	}
+	return syncTarget, nil
+}
+
+// getResourcesForPermission get all resource to sync from syncTarget status and resources flags. It is used to generate the rbac on
+// physical cluster for syncer.
+func (o *SyncOptions) getResourcesForPermission(ctx context.Context, config *rest.Config, syncTargetName string) (sets.String, error) {
+	kcpClient, err := kcpclient.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	// Poll synctarget to get all resources to sync, the ResourcesToSync set from the flag should be also added, since
+	// its related APIExport will not be generated until the syncer is started.
+	expectedResourcesForPermission := sets.NewString(o.ResourcesToSync...)
+	// secrets and configmaps are always needed.
+	expectedResourcesForPermission.Insert("secrets", "configmaps")
+	err = wait.PollImmediateWithContext(ctx, 100*time.Millisecond, 30*time.Second, func(ctx context.Context) (bool, error) {
+		syncTarget, err := kcpClient.WorkloadV1alpha1().SyncTargets().Get(ctx, syncTargetName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil //nolint:nilerr
+		}
+
+		// skip if there is only kubernetes export in the synctarget workspace, since we may not get syncedResources yet.
+		clusterName := logicalcluster.From(syncTarget)
+
+		if len(syncTarget.Spec.SupportedAPIExports) == 1 &&
+			syncTarget.Spec.SupportedAPIExports[0].Workspace.ExportName == "kubernetes" &&
+			(len(syncTarget.Spec.SupportedAPIExports[0].Workspace.Path) == 0 ||
+				syncTarget.Spec.SupportedAPIExports[0].Workspace.Path == clusterName.String()) {
+			return true, nil
+		}
+
+		if len(syncTarget.Status.SyncedResources) == 0 {
+			return false, nil
+		}
+		for _, rs := range syncTarget.Status.SyncedResources {
```

Comment:
We do not ensure compatibility at this stage because syncer is not started and resources have not be imported. The SyncedResources here is all resources from supportedExports updated by the server. I think it is tricky also to guarantee all resources from all supportedExports have been updated into the syncedResources, the controller  reads each APIExport and each APIResourceSchemas (and we ignore the non-existing ones). We might need a condition in synctarget status indicating all resources in supportedExports are updated to the syncedResources plus the lastObservedGeneration for the condition..

Reply:
So the resources imported from the SyncTarget (added through `--resource` and not present in the global `kubernetes` APIExport will not be taken in account for RBAC ?

Reply:
Sorry, I didn't look well. These are added from the command line option list above.
I assume it's all OK then.